### DISCUSSION TITLE
Filters and Columns Usability in Content Management

### DISCUSSION THREAD

**Discussion by luismts** [directus/directus#21051] [Feature Requests]
### Summary

This feature proposes enhancing the usability of the Content Management Section by allowing users to choose whether a filter or column should appear. This flexibility provides a more tailored and efficient experience for managing content within Directus.

Related: 

- **Filter usability** https://github.com/directus/directus/discussions/19707
- **Hiding fields in settings** https://github.com/directus/directus/issues/448

> You can see a full list of Enhancements and Desired Features for Directus [here](https://gist.github.com/luismts/ed40fb3ce57d0697f604a8ef59a2c875).

### Basic Example

```json
// Example Configuration
{
  "showFilters": true,
  "showColumns": true
}
```

### Motivation

The motivation behind this feature is to empower users with the ability to customize their Content Management Section based on individual workflows. By allowing users to selectively display filters and columns, we aim to improve user efficiency, reduce clutter, and enhance the overall user experience. This feature aligns with Directus' commitment to adaptability and user-centric design.

### Filters Configuration

- Introduce a configuration option to show/hide filters in the Content Management Section.
- Users can choose whether to display filters based on their relevance to the content being managed.
- Provide a user-friendly interface for configuring and saving filter visibility preferences.

### Columns Configuration

- Implement a configuration setting to show/hide columns in the Content Management Section.
- Allow users to personalize their view by selecting the columns they want to see.
- Provide an intuitive interface for managing and persisting column visibility preferences.

### Global Configuration

- Enable users to set a global preference for filters and columns visibility across different content types.
- Allow for a quick toggle between different configurations for various use cases.

### Must Have:

- Configurable options for showing/hiding filters and columns.
- Intuitive user interface for configuring and saving preferences.

### Should Have:

- Granular control over individual filter and column visibility.
- Global configuration options for consistency across content types.

### Could Have:

- Advanced customization options for filter and column appearance.

### Won't Have:
- Direct integration with every conceivable user interface customization.

### Drawbacks

- Increased code complexity in managing dynamic visibility configurations.
- Potential learning curve for users new to customizing interface elements.

### Alternatives

- Users can continue to use the default configuration without customization.
- The development of third-party plugins or extensions for interface customization.

### Adoption Strategy

- Provide clear documentation and tutorials for users to adopt and customize the new feature.
- Implement backward compatibility where possible to minimize breaking changes.
- Develop a migration path for existing Directus applications to accommodate the new configuration settings.

### Unresolved Questions

- Detailed design considerations for maintaining consistency in UI responsiveness.
- Exploration of additional customization options based on user feedback.
- Impact on performance with frequent toggling of filter and column visibility.

**Comment by luismts** [comment]
### Enhancements and Desired Features for Directus
I have created a thread with a list of features and enhancements that could further drive the utility and versatility of Directus.

You can see it [here](https://gist.github.com/luismts/ed40fb3ce57d0697f604a8ef59a2c875).

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
